docs/use_cases/retrieval_augmented_generation_eval.md

```diff
@@ -0,0 +1,181 @@
+
+<figure>
+<img src=/assets/use_cases/retrieval_augmented_generation_eval/rag_challenges.jpg alt="Classification of Challenges of RAG Evaluation" data-size="100" />
+<figcaption align="center">Slide from RAG Evaluation Presentation mentions <a href="https://arxiv.org/abs/2307.03172">'Lost in the Middle' problem</a></figcaption>
+</figure>
+
+For evaluation, we need metrics/methodology/frameworks that uses one or more
+tools. The tools that you use should cover each component of RAG. Such coverage
+ensures granular and thorough measurements.
+
+Evaluation metrics can assess:
+
+- Retrieval effectiveness
+- Coherence of generated responses
+- Relevance to the retrieved information.
+
+<img src=/assets/use_cases/retrieval_augmented_generation_eval/rag_granular.jpg alt="Granular Levels of Evaluation of RAG" data-size="100" />
+
+### Let's dive in !!
+
+## Strategies for Evaluation
+
+Now that we have defined different challenges and levels at which we can break down the RAG evaluation, let us zoom into the levels individually. 
+
+### Model Evaluation 
+
+To evaluate our model, we start with this [Massive Text Embedding Benchmark](https://huggingface.co/spaces/mteb/leaderboard). 
+We want to ensure that the model can understand the data that we encode.
+The benchmark shown above leverages different public/private datasets to evaluate and report on the different capabilities of individual models.
+If you're working with specialized domains, you may want to put together a specialized dataset to teach the model.
+
+We could either leverage the results here (provided our model belongs to this list) or run relevant 'tasks' for our custom model (instructions provided in the github [link](https://github.com/embeddings-benchmark/mteb#leaderboard))
+
+```python
+import logging
+
+from mteb import MTEB
+from sentence_transformers import SentenceTransformer
+
+
+logging.basicConfig(level=logging.INFO)
+
+
+model_name = "average_word_embeddings_komninos"
+model = SentenceTransformer(model_name)
+evaluation = MTEB(task_langs=["en"])
+evaluation.run(model, output_folder=f"results/{model_name}", eval_splits=["test"])
+
+print("--DONE--")
+```
+
+### Data Ingestion Evaluation
+
+We first validate the model, and train it on the language of our domain. 
+We can then configure data ingestion into our semantic retrieval store aka vector store.
+Various vector databases offer index configurations to influence and enhance the retrieval quality, based on the supported index types (Flat , LSH , HNSW and IVF).
+One such example here to improve HNSW [retrieval-quality](https://qdrant.tech/documentation/tutorials/retrieval-quality/).
+
+To evaluate ingestion, we need to focus on related variables, such as:
+
+* **Chunk size** - represents the size of each segment. This depends on the token limit of our embedding model. It also has a substantial impact on the contextual understanding of the data. That impacts the precision, recall, and relevancy of our results.
```

Comment:
It also controls the granularity of the information in the chunks. Depends a lot on what type of queries we expect.